crowbar_framework/lib/crowbar/repository.rb

```diff
@@ -16,16 +16,46 @@
 
 module Crowbar
   class Repository
-    attr_reader :platform, :id, :registry
+    attr_reader :platform, :id, :config
 
     class << self
       def load!
-        @config = YAML.load_file(Rails.root.join("config/repos.yml"))
+        etc_yml = "/etc/crowbar/repos.yml"
+
+        @all_repos = YAML.load_file(Rails.root.join("config/repos.yml"))
+
+        if File.exists? etc_yml
+          # merge data from etc config file
+          begin
+            etc_repos = YAML.load_file(etc_yml)
+          rescue SyntaxError
+            # ok, let's live without it
+            etc_repos = {}
+          end
+
+          etc_repos.each do |platform, repos|
+            if @all_repos.key? platform
+              repos["repos"].each do |id, repo|
+                # for repos that exist in our hard-coded file, we only allow
+                # overwriting a subset of attributes
+                if @all_repos[platform]["repos"].key? id
+                  %w(url ask_on_error).each do |key|
+                    @all_repos[platform]["repos"][id][key] = repo[key] if repo.key? key
```

Comment:
Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)